jobs/cache_maintenance/src/cache_maintenance/delete_indexes.py

```diff
@@ -0,0 +1,24 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2023 The HuggingFace Authors.
+
+import glob
+import logging
+import os
+from datetime import datetime, timedelta
+
+from libcommon.storage import StrPath
+
+
+def delete_indexes(
+    duckdb_index_cache_directory: StrPath, subdirectory: str, file_extension: str, expired_time_interval_seconds: int
+) -> None:
+    logging.info("delete indexes")
+    indexes_folder = f"{duckdb_index_cache_directory}/{subdirectory}/**/*{file_extension}"
+    logging.info(f"looking for all files with pattern {indexes_folder}")
+    for path in glob.glob(indexes_folder, recursive=True):
+        last_access_time_value = os.path.getatime(path)
+        last_access_datetime = datetime.fromtimestamp(last_access_time_value).replace(tzinfo=None)
+        now = datetime.now().replace(tzinfo=None)
```

Comment:
nit: maybe set this at the beginning of the function

Reply:
Done in https://github.com/huggingface/datasets-server/pull/1536/commits/f38ba31e3b7664ee3322416f71f07444cfddf906